pkg/api/util/ipc_endpoint_test.go

```diff
@@ -0,0 +1,212 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package util
+
+import (
+	"encoding/json"
+	"io"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"strings"
+	"testing"
+
+	pkgconfigmodel "github.com/DataDog/datadog-agent/pkg/config/model"
+	"github.com/stretchr/testify/assert"
+)
+
+func createConfig(t *testing.T, ts *httptest.Server) pkgconfigmodel.Config {
+	conf := pkgconfigmodel.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+
+	// create a fake auth token
+	authTokenFile, err := os.CreateTemp("", "")
+	assert.NoError(t, err)
+	authTokenPath := authTokenFile.Name()
+	os.WriteFile(authTokenPath, []byte("0123456789abcdef0123456789abcdef"), 0640)
+
+	addr, err := url.Parse(ts.URL)
+	assert.NoError(t, err)
+	localHost, localPort, _ := net.SplitHostPort(addr.Host)
+
+	// set minimal configuration that IPCEndpoint needs
+	conf.Set("auth_token_file_path", authTokenPath, pkgconfigmodel.SourceAgentRuntime)
+	conf.Set("cmd_host", localHost, pkgconfigmodel.SourceAgentRuntime)
+	conf.Set("cmd_port", localPort, pkgconfigmodel.SourceAgentRuntime)
+
+	return conf
+}
+
+func TestNewIPCEndpoint(t *testing.T) {
+	conf := pkgconfigmodel.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+
+	// create a fake auth token
+	authTokenFile, err := os.CreateTemp("", "")
+	assert.NoError(t, err)
+	authTokenPath := authTokenFile.Name()
+	os.WriteFile(authTokenPath, []byte("0123456789abcdef0123456789abcdef"), 0640)
+
+	// set minimal configuration that IPCEndpoint needs
+	conf.Set("auth_token_file_path", authTokenPath, pkgconfigmodel.SourceAgentRuntime)
+	conf.Set("cmd_host", "localhost", pkgconfigmodel.SourceAgentRuntime)
+	conf.Set("cmd_port", "6789", pkgconfigmodel.SourceAgentRuntime)
```

Comment:
This code is almost identical to `createConfig`. Would it be possible to reuse that helper function in this case?

Reply:
It was on purpose that these first two test cases don't use `createConfig`. The reason is that `createConfig` is meant to include everything needed for testing the http functionality of the IPCEndpoint, in particular it requires an `httptest.Server`. But simply calling the constructor `NewIPCEndpoint` without sending any requests doesn't need an http server; these two tests are exercising the constructor in isolation.